vignettes/na-notebook.Rmd

```diff
@@ -0,0 +1,510 @@
+---
+title: "na-notebook"
+output:
+  pdf_document: default
+  html_document: default
+date: "2024-08-19"
+---
+
+```{r setup, warning=FALSE, echo=FALSE}
+suppressPackageStartupMessages({
+  library(epidatr)
+  library(epiprocess)
+  library(dplyr)
+  library(tidyr)
+  library(ggplot2)
+})
+utils::globalVariables(c("is_na", "signal", "issue"))
+```
+
+# How Different NAs Arise and Strategies for Handling Them
+
+### NAs from merging
+
+First let's start with discussing the most common type of missing values that appeared in the context of my auxiliary signal project. When working with multiple signals each signal will likely begin recording at different times. In other words each signal's first data point $t_0$ will differ on the absolute time scale. As a result, when calling `epix_merge()` to combine multiple signals, the signals that started recording at a later point in time will have missing values for the time periods where the other signals were already recording. Here's a quick example
+
+Say signal A reads with the values (11, 12, 14, 15, 16) with its first recorded data point being October 10th. Signal B reads (20, 21, 22, 23, 24) with its first recorded data point being 2 days later on October 12th. And lastly signal C reads (30, 32, 33, 34) with its first recorded data point being 3 days after that on October 15th. Then when we merge all of the data points together it will look like this:
+
+| Time  | Signal A | Signal B | Signal C |
+|-------|----------|----------|----------|
+| $t_0$ | 11       | NA       | NA       |
+| $t_1$ | 12       | NA       | NA       |
+| $t_2$ | 13       | 20       | NA       |
+| $t_3$ | 14       | 21       | NA       |
+| $t_4$ | 15       | 22       | NA       |
+| $t_5$ | NA       | 23       | 30       |
+| $t_6$ | NA       | 24       | 32       |
+| $t_7$ | NA       | NA       | 33       |
+| $t_8$ | NA       | NA       | 34       |
+
+Where $t_0$ would be October 10th.
+
+Now let's take a look at some real data, and see how the same issue arises when combining Google Search data with doctor visits data for the states California, Florida, New York and Texas.
+
+```{r data_collection, warning=FALSE}
+# Gathering the Data
+states <- "ca,fl,ny,tx"
+today <- 20240725
+
+# s05 is Anosmia, Dysgeusia, Ageusia
+s05_smoothed_search <- pub_covidcast(
+  source = "google-symptoms",
+  signals = "s05_smoothed_search",
+  time_type = "day",
+  geo_type = "state",
+  time_values = epirange(20200220, today),
+  geo_values = states,
+  issues = epirange(20201130, today)
```

Comment:
suggest:
move to `time_values = "*", issues = epirange(12340101, today)` or some other absurdly early start issue